### ISSUE TITLE
Replace `isInSourceDirectories` with `directory : Metadata -> Directory`

### ISSUE THREAD

**Issue by lue-bird** [jfmengels/elm-review#123]
```elm
isInSourceDirectories : Metadata -> Bool
```
[isInSourceDirectories](https://dark.elm.dmy.fr/packages/jfmengels/elm-review/latest/Review-Rule?source=#isInSourceDirectories) only seems useful to check whether it's a test or not.
More useful would be
```elm
type Directory
    = TestDirectory
    | SourceDirectory String

directory : Metadata -> Directory -- or simply the path `String`
```
Allowing functionality similar to for example [`filterErrorsForFiles`](https://dark.elm.dmy.fr/packages/jfmengels/elm-review/latest/Review-Rule#filterErrorsForFiles) –
checking the directory origin _while still building the review rule_ to treat modules differently based on directory

**Comment by jfmengels** [comment]
Can you give some examples of what you'd do with this, or wish to do? I'm not sure I see what scenarios this enables that `isInSourceDirectories` can't already help with.

**Comment by lue-bird** [comment]
Seeing whether you are in the default `src` directory or in a private `package/name` directory or in a `test` directory or in a `generated` directory, etc.

Concrete example: I'm building a rule that should be able to not allow any file from within a package directory to import from outside of it

**Comment by jfmengels** [comment]
I recently added [`withFilePath`](https://elm-doc-preview.netlify.app/Review-Rule?repo=jfmengels%2Felm-review&version=master#withFilePath) (in `master`, not published yet), which gives you the relative path from `elm.json`.

Would that suit your need?

**Comment by lue-bird** [comment]
awsome

**Comment by jfmengels** [comment]
This has been released in v2.8.0.
